src/proxy.ts

```diff
@@ -0,0 +1,36 @@
+import { Request } from "express";
+
+export interface ResolvedURLInfo {
+  scheme: string;
+  host: string;
+}
+
+export function resolveProxyHeaders(req: Request): ResolvedURLInfo {
+  const scheme = getScheme(req);
+
+  return {
+    scheme: scheme != "" ? scheme : req.protocol,
+    host: (req.headers["x-forwarded-host"] as string) ?? req.headers.host,
+  };
+}
+
+function getScheme(req: Request): string {
+  let scheme: string = "";
+
+  if (req.headers["x-forwarded-proto"]) {
+    scheme = (req.headers["x-forwarded-proto"] as string).toLowerCase();
+  } else if (req.headers["x-forwarded-scheme"]) {
+    scheme = (req.headers["x-forwarded-scheme"] as string).toLowerCase();
+  } else if (req.headers["forwarded"]) {
+    const forwarded = req.headers["forwarded"] as string;
```

Comment:
Will the request only ever have one of these three headers? Trying to understand exactly what this change is doing and what we were doing before

Reply:
So before we were just taking whatever "url" the server received from the incoming request. But if a server is behind a reverse proxy that url will likely point to an IP address instead of the original domain name the request was sent to.

The headers I am looking at here are the standard headers used by most reverse proxies for transmitting the original data to the underlying server, so we are using it here to populate the url with the original information

Reply:
Awesome thanks, makes sense